x/ecocredit/server/core/query_class_issuers.go

```diff
@@ -0,0 +1,44 @@
+package core
+
+import (
+	"context"
+	"github.com/cosmos/cosmos-sdk/orm/model/ormlist"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ecocreditv1beta1 "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1beta1"
+	"github.com/regen-network/regen-ledger/x/ecocredit/v1beta1"
+)
+
+// ClassIssuers returns a list of addresses that are allowed to issue batches from the given class.
+func (k Keeper) ClassIssuers(ctx context.Context, request *v1beta1.QueryClassIssuersRequest) (*v1beta1.QueryClassIssuersResponse, error) {
+	pg, err := GogoPageReqToPulsarPageReq(request.Pagination)
+	if err != nil {
+		return nil, err
+	}
+
+	classInfo, err := k.stateStore.ClassInfoStore().GetByName(ctx, request.ClassId)
+	if err != nil {
+		return nil, err
+	}
+
+	it, err := k.stateStore.ClassIssuerStore().List(ctx, ecocreditv1beta1.ClassIssuerClassIdIssuerIndexKey{}.WithClassId(classInfo.Id), ormlist.Paginate(pg))
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Any benefit in converting the request id to uint64 rather than getting classInfo from store?

Reply:
hmm not sure im following, could you explain the suggesting a bit more?

Reply:
hmm not sure im following, could you explain the suggesting a bit more?

Reply:
We only need the id from class info, which we already have in the request. Do we need to get class info from state or can we just convert the id in the request?

Reply:
Ahhh gotcha, so the ClassId from the request is actually a string "C01", "BIO1", etc. so i use that to get the Row ID to query the issuer store

Reply:
Ah, I see. Makes sense as is.